docs/developer/stock-allocation.mdx

```diff
@@ -0,0 +1,298 @@
+---
+title: Stock allocation
+---
+
+## Introduction
+
+This guide explains how the allocations are handled in Saleor.
+
+## Terms introduction
+
+The warehouses, shipping zones, and channels must be properly linked together
+to provide the expected allocation. Below you can find the explanation of the terms
+that will be required to properly understand this guide:
+
+- channel - allow configuring your shop differently for different currencies;
+  go [here](./channels.mdx) for more information
+- shipping zone - allow for defining the cost of shipping in different areas of the world,
+  shipping zone contains shipping methods that can be used for a given part of the world
+- warehouse - contains stocks of specific products, assigned to a given channel and shipping zone,
+  can be defined as the place for click and collect
+- stock - corresponds to resources of specific product variant in the given warehouse
+- allocation - corresponds to the currently allocated stocks for a given order line
+
+The relations between the mentioned entities are shown in the below chart:
+
+```mermaid
+classDiagram
+    Channel "*" -- "*" ShippingZone
+    ShippingZone "1" -- "*" ShippingMethod
+    Channel "*" -- "*" Warehouse
+    Warehouse "*" -- "*" ShippingZone
+    Warehouse "1" -- "*" Stock
+    Stock "*" -- "1" ProductVariant
+    Stock "1" -- "*" Allocation
+    Allocation "*" -- "1" OrderLine
+    class Stock{
+      quantity: Number
+      quantity_allocated: Number
+    }
+    class Allocation{
+      quantity_allocated: Number
+    }
+    class Warehouse{
+      click_and_collect_option: String
+    }
+```
+
+## Allocation flow
+
+Stocks are allocated in the checkout completing process, for all lines
+that have track inventory turned on. The allocation is made for every line
+on stocks with the highest quantity available value. When there is not enough
+available quantity in one stock, multiple allocations are created. In case there is
+no enough quantity in all product stocks - the `InsufficientStock` error is raised.
+
+The line can be allocated only in the stocks that fulfill the following conditions:
+
+- the stock's warehouse is available in the channel in which the order is created
+- the stock's warehouse has a shipping zone that includes the shipping address
+  (or billing address if the shipping address is not set) country code
+  and this shipping zone is available in the channel in which the order is created
+- the stock has available quantities
```

Comment:
**Comment**:
This is super helpful. In the future, it would be nice to enrich these kinds of descriptions with screenshots.

Reply:
Do you mean dashboard screenshots? This might be a good idea, although dev docs are more API-oriented. We could also link to dashboard sections here.

Reply:
Something to discuss probably as it should be unified across other guides (maybe as a separate task).

Reply:
> Do you mean dashboard screenshots? This might be a good idea, although dev docs are more API-oriented.

Now that I think about it, I think you are right. As far as I can see, we don't have dashboard screenshots in the "Developer's Guide" right now, and that's because of the clear line between what those two tutorials are. Maybe you are deep in on some technical problems, and rolled your own dashboard, so why are you seeing the screenshots of the Saleor dashboard?

<thinking_out_loud>

I am just thinking about my experience of starting with Saleor: sometimes I got lost in the "Dashboard Manual", so I was looking for answers in the "Developer's Guide". That is not necessarily a reason to add Dashboard screenshots there, but for sure to meticulously reference sections from "Developer's Guide" inside "Dashboard Manual".

</thinking_out_loud>

Reply:
@peelar I really like a visual aid, but one thing to keep in mind about screenshots is that they get dated, and become a source of confusion.